bin/propolis-server/src/lib/server.rs

```diff
@@ -173,6 +174,21 @@ impl VmControllerState {
     }
 }
 
+/// Objects related to Propolis's Oximeter metric production.
+pub struct OximeterState {
+    /// A task spawned at initial startup to register with Nexus for metric
+    /// production.
+    registration_task: Mutex<Option<tokio::task::JoinHandle<()>>>,
+
+    /// The host task for this Propolis server's Oximeter server.
+    server_task: Mutex<Option<tokio::task::JoinHandle<()>>>,
+
+    /// The metrics wrapper for "server-level" metrics, i.e., metrics that are
+    /// tracked by the server itself (as opposed to being tracked by a component
+    /// within an instance).
+    stats: Mutex<Option<crate::stats::ServerStatsOuter>>,
```

Comment:
Do these need to be accessed independently, or would it make sense to instead wrap the whole `OximeterState` in a mutex in `ServiceProviders`?

Reply:
Yeah, I was not sure either. The stats might need to be accessed independently, but they're the only thing I expect to be accessed with any frequency. I'll try out hosting the mutex up a layer.

Reply:
Done in e295e14